### DISCUSSION TITLE
Can't download anymore (antivirus issue)

### DISCUSSION THREAD

**Discussion by tubefan** [aclap-dev/video-downloadhelper#14] [Help and Q&A]
### Your question

I'm using Video DownloadHelper version 8.2.0.17a1 (beta) and suddenly I can't download from YOUTUBE anymore (f. e.  https://www.youtube.com/watch?v=_mEC54eTuGw)

**Comment by GrampaWildWilly** [comment]
Your problem report is faulty.  You need to provide the version numbers of your software.  You mentioned only your version of VDH.  That's not enough.

I went to your Stones video & tried to get VDH to download the 2048x1152 webm variant.  But that failed with an error that is well known & is already under investigation.

I then downloaded the 1920x1080 mp4 variant without incident.  It took 4 seconds to download.  The resulting file is size 94.4M, duration 3:45, resolution 1920x1080, video bitrates in the 3300-3500kbps range (rather better than normal for YouTube, although still not very good), 25fps, stereo audio at sampling rate 44.1kHz & bitrate 127kbps.  It played fine in VLC with both video & audio right to the end.

Now here's the part that should be in every problem report:

I am running Windows 11 Pro 64-bit version 23H2 build 22631.3085, Firefox 122.0 64-bit, licensed VDH 8.2.0.17a1 beta, CoApp 2.0.14 beta.

Yes, a beta of the CoApp.  Look around at other threads in this forum & you'll discover an announcement of sorts of a beta for the CoApp.  I found it by accident.  You can, too.

Here's a hint:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/409b9dcc-f1aa-49b1-8e47-76187d9f8d2c)

**Reply by tubefan** [reply]
OK, I'm using Windows 10, 64bit-version  22H2, Built 19045 3930, Firefox 122.0 64bit, licensed VDH 8.2.0.17 a1 beta, VdhCoApp 2.0.14 
SORRY, but I don't find the CoApp-BETA-Version! Could you give a link?

I get this download error message:**

SideDownload error:

[tcp @ 0000015265d8cf00] Connection to tcp://manifest.googlevideo.com:443 failed: Error number -138 occurred

[in#0 @ 0000015264159cc0] Error opening input: Error number -138 occurred

Error opening input file https://manifest.googlevideo.com/api/manifest/hls_playlist/expire/1706652104/ei/aB25ZavdFJ6J6dsPi8mkqAo/ip/2003:df:7f4f:2c00:d05:34f:9faf:1d80/id/fe6102e78793b86c/itag/270/source/youtube/requiressl/yes/ratebypass/yes/pfa/1/sgovp/clen%3D95282405%3Bdur%3D225.640%3Bgir%3Dyes%3Bitag%3D137%3Blmt%3D1705987455877047/rqh/1/hls_chunk_host/rr4---sn-h0jeened.googlevideo.com/xpc/EgVo2aDSNQ%3D%3D/mh/ie/mm/31,29/mn/sn-h0jeened,sn-h0jelne6/ms/au,rdu/mv/m/mvi/4/pl/42/force_finished/1/gcr/de/initcwndbps/1762500/vprv/1/playlist_type/DVR/dover/13/txp/4535434/mt/1706630032/fvip/2/short_key/1/keepalive/yes/fexp/24007246/sparams/expire,ei,ip,id,itag,source,requiressl,ratebypass,pfa,sgovp,rqh,xpc,force_finished,gcr,vprv,playlist_type/sig/AJfQdSswRgIhAPScQ_CiEGbvu-J7g1SFMhgp8MwBlYUCiAN57eNTXExYAiEA6Fom4g3coR7fVYw5O7FFga4kF7S0lmexAtbGnqEKXSk%3D/lsparams/hls_chunk_host,mh,mm,mn,ms,mv,mvi,pl,initcwndbps/lsig/AAO5W4owRQIgF8rJ-DGI8LAu669AudlcD3xKHNEM4rt3TCF_LibtwQECIQCc6eyAL-MUb9be9bA4bQwCXhXE0zZ_OgsW0DDPBwg_Uw%3D%3D/playlist/index.m3u8.

Error opening input files: Error number -138 occurred

**Comment by GrampaWildWilly** [comment]
It's the discussion about no sound or sound dropping.  But it's closed.  That's why you need to filter on All.  The default filter shows only open discussions.  I've asked for the CoApp betas to have their own link in the CoApp area of GitHub.  At the moment, there's no link in a readily accessible location.  There's only a mention buried in a discussion of sound problems.

As for the error you're getting, it doesn't mean anything to me.  Maybe if I were a dev it would help but I'm just another user.  I don't work for VDH.  All I can tell you is the Stones video downloaded fine for me.  But most stuff on YouTube has been downloading fine for me, even before I got the CoApp beta.

I think it's weird that your error mentions Side Download.  The new VDH 8.2.0.17a1 beta doesn't do Side Downloads any more.  There's just Downloads.  When you installed this beta, did you get a page saying congratulations & listing all the new things in the beta?  If you didn't, maybe the install didn't complete properly.  I've had that issue in the past.  You might try installing it again, possibly after uninstalling it.  You should export your settings, then uninstall, then reinstall, then import your settings.  That way, you won't need to redo all your settings.  But if you did get that congratulations page, then I believe you should be good to go & uninstall/reinstall would be pointless.

**Reply by tubefan** [reply]
Thank you very much, but unfortunately that doesn't solve my problem.
Can I please get an answer/solution directly from VDH people?

**Comment by paulrouget** [comment]
I see `tcp://manifest.googlevideo.com`. That's surprising.

Can you take a screenshot of the item you click on?

Before starting the download, can you go to the submenu, click `details` and tell me what you see for masterManifest / audioMediaManifest / videoMediaManifest?

**Reply by tubefan** [reply]
Hi, here it is:

audioMediaManifest | https://manifest.googlevideo.com/api/manifest/hls_playlist/expire/1706724338/ei/kje6ZZO7Bu6Ai9oPjYKRiAc/ip/2003:df:7f4f:2c00:a50f:7210:bb58:28c3/id/fe6102e78793b86c/itag/234/source/youtube/requiressl/yes/ratebypass/yes/pfa/1/goi/133/sgoap/clen%3D3653754%3Bdur%3D225.721%3Bgir%3Dyes%3Bitag%3D140%3Blmt%3D1705984878829028/rqh/1/hls_chunk_host/rr2---sn-h0jelne6.googlevideo.com/xpc/EgVo2aDSNQ%3D%3D/mh/ie/mm/31,26/mn/sn-h0jelne6,sn-4g5e6nze/ms/au,onr/mv/m/mvi/2/pl/42/force_finished/1/gcr/de/initcwndbps/1640000/vprv/1/playlist_type/DVR/dover/13/txp/4532434/mt/1706702451/fvip/2/short_key/1/keepalive/yes/fexp/24007246/sparams/expire,ei,ip,id,itag,source,requiressl,ratebypass,pfa,goi,sgoap,rqh,xpc,force_finished,gcr,vprv,playlist_type/sig/AJfQdSswRQIhAKll-eHS6bJDfyI8fxPNq1M0BBpEONP42Fa8zLT5TeInAiAqi4cu8WnhNFrfjeSWDjyebTIO2-VU2NR6FWsEYvfQDQ%3D%3D/lsparams/hls_chunk_host,mh,mm,mn,ms,mv,mvi,pl,initcwndbps/lsig/AAO5W4owRgIhANt1DmKwQu_Fqgwb6-8iW1axz2ySnRnzzLeddw3tShOjAiEAk2Vomzs9nCE3OZ9HWQsuVbzZsvxELBnKpiM4lnV_WOY%3D/playlist/index.m3u8

masterManifest | https://manifest.googlevideo.com/api/manifest/hls_variant/expire/1706724338/ei/kje6ZZO7Bu6Ai9oPjYKRiAc/ip/2003%3Adf%3A7f4f%3A2c00%3Aa50f%3A7210%3Abb58%3A28c3/id/fe6102e78793b86c/source/youtube/requiressl/yes/xpc/EgVo2aDSNQ%3D%3D/playback_host/rr2---sn-h0jelne6.googlevideo.com/mh/ie/mm/31%2C26/mn/sn-h0jelne6%2Csn-4g5e6nze/ms/au%2Conr/mv/m/mvi/2/pl/42/hfr/1/force_finished/1/demuxed/1/tts_caps/1/maudio/1/gcr/de/initcwndbps/1640000/vprv/1/go/1/mt/1706702451/fvip/2/nvgoi/1/short_key/1/ncsapi/1/keepalive/yes/fexp/24007246/dover/13/itag/0/playlist_type/DVR/sparams/expire%2Cei%2Cip%2Cid%2Csource%2Crequiressl%2Cxpc%2Chfr%2Cforce_finished%2Cdemuxed%2Ctts_caps%2Cmaudio%2Cgcr%2Cvprv%2Cgo%2Citag%2Cplaylist_type/sig/AJfQdSswRQIgDcqgv0DihEo3Yp2VOhuvBlS65mEG1yl2RZygfR_0J0UCIQDoDlpjYnRIOP_NKpnJmm48wCmUoYI_Hnyji9f9MIeVcg%3D%3D/lsparams/playback_host%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps/lsig/AAO5W4owRAIgSek2VGdDwxxsOPKNjxApGtSC65E_jHPHhrblEdZM5OICICTB3HJyUAFMWWRo4lt_9bpQkO1HbzurqdqTJr33QY12/file/index.m3u8

videoMediaManifest | https://manifest.googlevideo.com/api/manifest/hls_playlist/expire/1706724338/ei/kje6ZZO7Bu6Ai9oPjYKRiAc/ip/2003:df:7f4f:2c00:a50f:7210:bb58:28c3/id/fe6102e78793b86c/itag/270/source/youtube/requiressl/yes/ratebypass/yes/pfa/1/sgovp/clen%3D95282405%3Bdur%3D225.640%3Bgir%3Dyes%3Bitag%3D137%3Blmt%3D1705987455877047/rqh/1/hls_chunk_host/rr2---sn-h0jelne6.googlevideo.com/xpc/EgVo2aDSNQ%3D%3D/mh/ie/mm/31,26/mn/sn-h0jelne6,sn-4g5e6nze/ms/au,onr/mv/m/mvi/2/pl/42/force_finished/1/gcr/de/initcwndbps/1640000/vprv/1/playlist_type/DVR/dover/13/txp/4535434/mt/1706702451/fvip/2/short_key/1/keepalive/yes/fexp/24007246/sparams/expire,ei,ip,id,itag,source,requiressl,ratebypass,pfa,sgovp,rqh,xpc,force_finished,gcr,vprv,playlist_type/sig/AJfQdSswRQIgcYV5eYlYM_1zXhH5ZKCzHAjo35P3sNbjESrS5f6RnwMCIQDmOeJqcvWn6V-HkAHFCzza4oWxT_SRPcXTye5eQvaTUA%3D%3D/lsparams/hls_chunk_host,mh,mm,mn,ms,mv,mvi,pl,initcwndbps/lsig/AAO5W4owRQIhAIlRaZO4ziMG3AAnIZNN8OergSt8Uxs2q9GjvpmkiWPOAiAMmKU5QjHPqFGqzgtQcmxOu5z65Y3apSPEZVMGBkRxHQ%3D%3D/playlist/index.m3u8

**Comment by GrampaWildWilly** [comment]
I offer these for comparison.  I am able to download this video so maybe comparing what I got with whatever our original poster tells us will be helpful.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/a77d620e-1238-476e-a8aa-f222cf959067)

```
masterManifest | https://manifest.googlevideo.com/api/manifest/hls_variant/expire/1706704111/ei/j-i5ZYXfL83HybgPotCysA0/ip/2603%3A6011%3Ac306%3Aa341%3A1557%3A7f9%3A2754%3A9a52/id/fe6102e78793b86c/source/youtube/requiressl/yes/xpc/EgVo2aDSNQ%3D%3D/playback_host/rr3---sn-vgqsrn6z.googlevideo.com/mh/ie/mm/31%2C26/mn/sn-vgqsrn6z%2Csn-p5qddn7k/ms/au%2Conr/mv/m/mvi/3/pl/39/hfr/1/force_finished/1/demuxed/1/tts_caps/1/maudio/1/gcr/us/initcwndbps/1412500/vprv/1/go/1/mt/1706682052/fvip/5/nvgoi/1/short_key/1/ncsapi/1/keepalive/yes/fexp/24007246/dover/13/itag/0/playlist_type/DVR/sparams/expire%2Cei%2Cip%2Cid%2Csource%2Crequiressl%2Cxpc%2Chfr%2Cforce_finished%2Cdemuxed%2Ctts_caps%2Cmaudio%2Cgcr%2Cvprv%2Cgo%2Citag%2Cplaylist_type/sig/AJfQdSswRAIgTb41LBoFqZkBNzOTkEHzC6IHHJjr6NB12foM2gnuYZACIHlsKASxqnGVHbd1YZCltKqaR5hfdW1ndeGOQbO2TI4t/lsparams/playback_host%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps/lsig/AAO5W4owRgIhAOljAeTSoKsap3LLlm4g8uIB3IcjqUET_RiJo8HJNW0WAiEA2UXNkoe86KFoMfiT0W5wnaP-ia2W3UexQELj_Ia8ZSI%3D/file/index.m3u8

videoMediaManifest | https://manifest.googlevideo.com/api/manifest/hls_playlist/expire/1706704111/ei/j-i5ZYXfL83HybgPotCysA0/ip/2603:6011:c306:a341:1557:7f9:2754:9a52/id/fe6102e78793b86c/itag/270/source/youtube/requiressl/yes/ratebypass/yes/pfa/1/sgovp/clen%3D95282405%3Bdur%3D225.640%3Bgir%3Dyes%3Bitag%3D137%3Blmt%3D1705987455877047/rqh/1/hls_chunk_host/rr3---sn-vgqsrn6z.googlevideo.com/xpc/EgVo2aDSNQ%3D%3D/mh/ie/mm/31,26/mn/sn-vgqsrn6z,sn-p5qddn7k/ms/au,onr/mv/m/mvi/3/pl/39/force_finished/1/gcr/us/initcwndbps/1412500/vprv/1/playlist_type/DVR/dover/13/txp/4535434/mt/1706682052/fvip/5/short_key/1/keepalive/yes/fexp/24007246/sparams/expire,ei,ip,id,itag,source,requiressl,ratebypass,pfa,sgovp,rqh,xpc,force_finished,gcr,vprv,playlist_type/sig/AJfQdSswRAIgWPqcgWRMuD9KVwTw8BdrR1eNIurermHiPi88aPJdFd8CIGGocRfIKeSVTPICpA-bLj7yIbmh551mV7xKc5znppDN/lsparams/hls_chunk_host,mh,mm,mn,ms,mv,mvi,pl,initcwndbps/lsig/AAO5W4owRAIgdk_Orz5O_oyuK1_180Urb5g8Z3Q7cZ9USniJlfl6xbkCIG4UQ0d6dV_gQwauefQYUxibiqh83O_aYGwy55agTKVi/playlist/index.m3u8

audioMediaManifest | https://manifest.googlevideo.com/api/manifest/hls_playlist/expire/1706704111/ei/j-i5ZYXfL83HybgPotCysA0/ip/2603:6011:c306:a341:1557:7f9:2754:9a52/id/fe6102e78793b86c/itag/234/source/youtube/requiressl/yes/ratebypass/yes/pfa/1/goi/133/sgoap/clen%3D3653754%3Bdur%3D225.721%3Bgir%3Dyes%3Bitag%3D140%3Blmt%3D1705984878829028/rqh/1/hls_chunk_host/rr3---sn-vgqsrn6z.googlevideo.com/xpc/EgVo2aDSNQ%3D%3D/mh/ie/mm/31,26/mn/sn-vgqsrn6z,sn-p5qddn7k/ms/au,onr/mv/m/mvi/3/pl/39/force_finished/1/gcr/us/initcwndbps/1412500/vprv/1/playlist_type/DVR/dover/13/txp/4532434/mt/1706682052/fvip/5/short_key/1/keepalive/yes/fexp/24007246/sparams/expire,ei,ip,id,itag,source,requiressl,ratebypass,pfa,goi,sgoap,rqh,xpc,force_finished,gcr,vprv,playlist_type/sig/AJfQdSswRQIhAIV1AfZVKB7jrSnOLBZVOT-phmbPavsEZYz2s75JrZV5AiBn2s9w5d4pGuWkcNrHMZB2w4dBOOi7TbnsZePfrqi7Xw%3D%3D/lsparams/hls_chunk_host,mh,mm,mn,ms,mv,mvi,pl,initcwndbps/lsig/AAO5W4owRQIgejkO50IOQeU_7wiCCTUmo7KsH32TXpDvSwjHM08UNEQCIQCrrXjQQhtqU6YqLySWQK9Sp4UiJlYeQ9fCgfWWx-GtoQ%3D%3D/playlist/index.m3u8
```

YouTube sure knows how to make ugly URLs.

**Reply by tubefan** [reply]
I also get the error message with the MP4 setting. (I had tried both settings.)

**Reply by GrampaWildWilly** [reply]
Setting?  What VDH setting are you talking about?  Or do you just mean selection on the menu?  Those are generally referred to as variants.  The settings are elsewhere.

Meanwhile, it would be really helpful if you provided the information Paul asked for.  You can use my response above as a model for what he is looking for from you.

**Reply by tubefan** [reply]
Sorry, I meant the menu selection (variant). 
I just replied to Paul (see there).

**Comment by paulrouget** [comment]
> I also get the error message with the MP4 setting. (I had tried both settings.)

Interesting. Thank you.

Can you please do this:

- take a screenshot of the item you click on
- before starting the download, can you go to the submenu (the arrow on the right) click details and tell me what you see for masterManifest / audioMediaManifest / videoMediaManifest?

**Reply by tubefan** [reply]
Hi Paul, here is the screenshot.

![Screenshot (5)](https://github.com/aclap-dev/video-downloadhelper/assets/158179976/f9049d00-b7ff-4d79-a3bd-cea38b3bae3c)

**Comment by paulrouget** [comment]
You confirm you're using the CoApp 2.0.14?

Can you try in [a private window](https://www.mozilla.org/en-US/firefox/features/private-browsing/) see if you get a different result?

**Reply by tubefan** [reply]
It also doesn't work in private mode.
Yes, I'm using CoApp 2.0.14.

--------------------------------------

SideDownload error:

[tcp @ 000001ea3409c900] Connection to tcp://manifest.googlevideo.com:443 failed: Error number -138 occurred

[in#0 @ 000001ea34099e40] Error opening input: Error number -138 occurred

Error opening input file https://manifest.googlevideo.com/api/manifest/hls_playlist/expire/1706789545/ei/STa7ZdavF9aF6dsP5ceMsA8/ip/139.28.177.98/id/fe6102e78793b86c/itag/270/source/youtube/requiressl/yes/ratebypass/yes/pfa/1/sgovp/clen%3D95282405%3Bdur%3D225.640%3Bgir%3Dyes%3Bitag%3D137%3Blmt%3D1705987455877047/rqh/1/hls_chunk_host/rr2---sn-4g5e6nze.googlevideo.com/xpc/EgVo2aDSNQ%3D%3D/mh/ie/mm/31,26/mn/sn-4g5e6nze,sn-aigl6nsk/ms/au,onr/mv/m/mvi/2/pl/24/force_finished/1/gcr/de/initcwndbps/596250/vprv/1/playlist_type/DVR/dover/13/txp/4535434/mt/1706767485/fvip/2/short_key/1/keepalive/yes/fexp/24007246/sparams/expire,ei,ip,id,itag,source,requiressl,ratebypass,pfa,sgovp,rqh,xpc,force_finished,gcr,vprv,playlist_type/sig/AJfQdSswRQIgdT0fg600loqCMrEQOhUlWUcCaQP5YFKMQj7po0z4jQYCIQDbjL1UdK15-Ov1ygo6eOhcLlj1PXaPdbE6Zo9JEjWocQ%3D%3D/lsparams/hls_chunk_host,mh,mm,mn,ms,mv,mvi,pl,initcwndbps/lsig/AAO5W4owRAIgQGR-bX-5hBB3wnMNGSQ5TB0_Se6FEUwDIm9cdBHkefQCIEejr-cHYOP-K-yBcd3GtGnWFKQDb3LRaPlzZ9l5pc25/playlist/index.m3u8.

Error opening input files: Error number -138 occurred

**Reply by paulrouget** [reply]
Ok - so it's a TCP level issue, which, to me, indicates a OS-level issues.

Questions:

- do you get the error right away, or it takes a few seconds before you see the error?
- do you use a VPN or a Proxy?
- Do you use an antivirus?

**Reply by tubefan** [reply]
It takes 2-3 seconds before the error.
I use BITDEFENDER but without VPN.

**Reply by GrampaWildWilly** [reply]
On a hunch, can you check your Bitdefender quarantine?

**Reply by tubefan** [reply]
There is no entry in quarantine.

**Reply by GrampaWildWilly** [reply]
I don't use Bitdefender but doesn't it have a firewall?  Is there some way to check a firewall log to see if it's blocking something to do with our CoApp?

**Reply by tubefan** [reply]
When I disable Bitdefender firewall, I get this message:

Convert2 error:
Output C:\Users\Besitzer\dwhelper\The Rolling Stones - Angry (Official Music Video)-1.mp4 same as Input #0 - exiting
Error opening output file C:\Users\Besitzer\dwhelper\The Rolling Stones - Angry (Official Music Video)-1.mp4.
Error opening output files: Invalid argument

**Reply by GrampaWildWilly** [reply]
Well, at least it did something different.  This tells me that you are using the default VDH directory dwhelper as the target of your download.  Nothing wrong with that.  The -1 at the end of the file name tells me you've tried to download this before during this browser session.  Nothing wrong with that either.  I assume that at some point since you started having this problem, you have closed all your browser windows & started a new Firefox session.  You're not trying to do a Download and Convert, are you?  The "Convert2 error" makes me ask that.  But the input file & output file being the same . . . that is too weird.

I'm getting desperate here.  Let's try this.  Go into the VDH settings, drop down the More... menu, execute Reload extension.  Go into that dwhelper directory & erase all files that have anything to do with the Stones video.  Go back to the YouTube page & reload it.  Try again with your firewall off.  By try again I mean do a Download, not a Quick Download, not a Download and Convert.  Just a Download.  I'm hoping you get a file selection dialog.  If it comes up with the dwhelper directory selected, just accept it.

I have to say I am completely baffled by this one.  I think Paul is, too, judging from his comments.

As for the 2.0.10 version, the one you asked about would be the one for your system.  But before we get you going back to an older version, let's keep trying to make the current version work for the time being.

**Reply by tubefan** [reply]
I think we are getting closer to a solution. GWW, I followed your suggestion. The download works, but ONLY when the BITDEFENDER firewall is turned off. Strange, this was NOT a problem until a few days ago and everything was going well under Bitdefender so far!?? I have now entered vdhcoapp.exe as an exception rule in the Bitdefender program, but this does not help: With Bitdefender activated (and despite this rule) no download is possible!
This leads to the problem that I don't want to turn off Bitdefender before every (Youtube-) download. (Any suggestions on this?) Or maybe it's still necessary to return to the 2.0.10 version after all? (Would it be this file: vdhcoapp-2.0.10-windows-x86_64-installer.exe)? 
**Guys, by the way, thank you so far for your effort and patience. GREAT JOB!!**

**Reply by GrampaWildWilly** [reply]
I think you need to make exceptions in the Bitdefender firewall for all the .exe files in your CoApp directory.  There's more than just vdhcoapp.exe.  That one is actually not the one you use if you are using the VDH beta.  The ones that are actually used are ffprobe.exe & most importantly ffmpeg.exe.  But in that directory.  If you have regular ffmpeg that you got from ffmpeg.org somewhere else on your system, I think it's possible you might have to make an exception in the Bitdefender firewall for that as well.  But I don't know about that.  Like I said, I don't use Bitdefender.  I've just kept the antivirus built into Windows 11.  It hasn't raised any flags on VDH or the CoApp since I started using W11 at the beginning of December, not quite 2 months ago.  So I haven't learned yet how to make exceptions in that.  And I hope I never have to.  But I'm sure I will.

This is all just a false positive on the part of Bitdefender.  A bunch of users got all wound up & angry a couple of weeks ago over in the Google forum.  But it's just an exception.  I imagine the Bitdefender developers will get around to correcting their false positive.  It is their error.  They're the ones throwing up the false positive.  VDH & more specifically the CoApp are known safe applications.  You've been using it for a while.  I've been using it for several years.  It's known to be safe.  This is all about the antivirus getting entirely too picky.

Reverting to CoApp 2.0.10 does not seem like it's our solution here.  That is the right installer program for your system for CoApp 2.0.10.  But I wouldn't go back to that.  We're almost there with the current version.

**Reply by GrampaWildWilly** [reply]
You should probably make both the antivirus & the firewall exceptions apply to the whole CoApp directory.  That would probably simplify things.

**Reply by tubefan** [reply]
I've added ffprobe.exe and ffmeg.exe to my Bitdefender exception rules and the downloads seem to work now (wow). BUT now the following problem: After a successful Youtube download, I can't repeat the same thing again (i.e. download the SAME video again) because after starting DH, the functions don't fully appear in the drop-down menu. There is only to be seen: Open Media, Open Folder, Delete, Details, Copy URL, Use this action as default (I hope I've translated it correctly from German to English.) So, there is no list to select Quick Download, Download, Quick audio-only download, audio-only download, Download and Convert, etc.
To do the same task again (download THE SAME video again) and get the complete drop-down menu again, it's not enough to open the YOUTUBE page again, I even have to completely restart FIREFOX.  (The download for more/new videos works!) Is there a solution for this?

**Reply by GrampaWildWilly** [reply]
![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/8dd3c38e-a559-4d8c-ad47-b6b6651836f6)

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/7fea01ef-6df3-4125-bda1-d5363281036e)

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/24a3dec8-87dd-453c-9be1-71c8def0c3c0)

![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/7f5339dc-c8f1-4054-a287-ef3295859780)

That beta is in the new standard place:

https://github.com/aclap-dev/video-downloadhelper/wiki/Beta-versions

Remember that link.  It's going to come in handy.

**Reply by GrampaWildWilly** [reply]
By the way, for future reference, you should try this before you close all browser windows:

![#05](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/2d8e7d8d-d509-4fc8-afe3-e118b94e3347)

Know that when you execute this function, it loads fresh copies of the VDH browser extension & the CoApp.  If you have any downloads in progress, they will be abruptly terminated, most likely leaving behind damaged, unusable files.  If you have a download queue, it will be emptied.  You would have to redo all of those downloads from the beginning.

This function will probably be enough in most cases.  But sometimes it might not be enough.  I would try it before closing all browser windows.  If you're to the point of considering closing all browser windows anyway, it can't hurt anything to try this first.  If it doesn't work, then go ahead & cycle the browser.

**Reply by GrampaWildWilly** [reply]
My eyes might be playing tricks on me but I could swear I saw Bill Wyman in that video.  There seemed to be a lot of footage of the very much younger versions of the guys, from at least 40 years ago.  There's one point where it shows Keith sticking a cigarette in his mouth.  Seems to me I read recently that he had quit smoking.

**Comment by paulrouget** [comment]
If the antivirus doesn't say anything about blocking the coapp connection:

Can you try this:

- in the VDH settings: `More > Video Qualities`, uncheck `Ignore low quality videos`
- you'll see an additional download option in youtube (one that stands on its own); can you try to download that one see if it makes any difference?

**Reply by tubefan** [reply]
Changing the video quality setting didn't help. 
Can you please explain to me where/how to find the mentioned download option in youtube

**Reply by paulrouget** [reply]
Before changing the setting:

<img width="496" alt="Screenshot 2024-02-01 at 16 06 28" src="https://github.com/aclap-dev/video-downloadhelper/assets/373579/cb9cd313-01f3-4b9a-abd5-08fb1b0a00e2">

After changing the setting:

<img width="494" alt="Screenshot 2024-02-01 at 16 06 44" src="https://github.com/aclap-dev/video-downloadhelper/assets/373579/232bb469-b137-45f5-b35e-574f151f858a">

See the extra download at the bottom?

Do you see it?
And do you get the same error when downloading with that new download option?

**Reply by tubefan** [reply]
I get the same error.
Shall I try & download the proposed coapp version 2.0.10.
Which one (vdhcoapp-2.0.10-windows-x86_64-installer.exe) ?

**Reply by paulrouget** [reply]
I'm very surprised you're getting the same error.

You're also getting `manifest.googlevideo.com:443 failed: Error number -138 occurred` ?

**Reply by tubefan** [reply]
After changing the video quality setting I get this:

Exit code: 1

moz-extension://df58ea19-061b-4f48-b0d8-cf97a00f13dc/background/main.js:1
receive@moz-extension://df58ea19-061b-4f48-b0d8-cf97a00f13dc/background/main.js:1:13157
Jp</callCatchAppNotFound/</<@moz-extension://df58ea19-061b-4f48-b0d8-cf97a00f13dc/background/main.js:1:42562

Shall I try & download the proposed coapp version 2.0.10.
Which one (vdhcoapp-2.0.10-windows-x86_64-installer.exe) ?

**Reply by paulrouget** [reply]
You can give it a try.

But that error you just shared is really strange.

In the addon setting, there's a copy to clipboard option. Can you please copy it, and share it here?

<img width="827" alt="Screenshot 2024-02-01 at 17 13 32" src="https://github.com/aclap-dev/video-downloadhelper/assets/373579/e3913ba1-4be7-429a-9380-3689a50d5b4e">

**Reply by tubefan** [reply]
clipboard entry:

version 8.2.0.17a1
channel: beta
built date: 2024-01-30

**Reply by paulrouget** [reply]
Oh, that's not what I was expecting.

The settings say the coapp is not recognized?

**Reply by tubefan** [reply]
OK, what to do?
Install  **_vdhcoapp-2.0.10-windows-x86_64-installer.exe_**  ??

**Reply by paulrouget** [reply]
Does the settings say that the CoApp is not recognized?

Here, what does it say?

<img width="893" alt="Screenshot 2024-02-01 at 18 23 37" src="https://github.com/aclap-dev/video-downloadhelper/assets/373579/1825da5e-491b-4eeb-aa1b-e255af960b05">

**Reply by tubefan** [reply]
⊖
Video DownloadHelper 8.2.0.17a1 (beta)
Video DownloadHelper
Version 8.2.0.17a1 (beta)
Browsersprache: de
Produktionsversion
Kompilierungszeitpunkt: 2024-01-30
Kompilierungsoptionen: browser=firefox, linuxlic=false, noyt=false, target=mozilla
⊕
Plattform
⊖
Begleitanwendung installiert
****Begleitanwendung gefunden**: VdhCoApp 2.0.14**
Pfad zur Begleitanwendung: C:\Program Files\DownloadHelper CoApp\vdhcoapp.exe

⊕
Lizenz geprüft

**Reply by GrampaWildWilly** [reply]
On Windows, that would look like this:

Companion App installed
Found companion app: VdhCoApp 2.0.14
Companion app binary: C:\DownloadHelper CoApp\vdhcoapp.exe

The exact directory could be different.

**Reply by GrampaWildWilly** [reply]
I assume Lizenz geprüft = License verified

**Reply by tubefan** [reply]
It said all the time: License verified. 
When I renewed it a minute ago, this message came up: 
License revoked (renew validity). Would you please reactivate it?

**Reply by paulrouget** [reply]
> C:\Program Files\DownloadHelper CoApp\vdhcoapp.exe

Ok, so, your coapp is recognized. This doesn't match your copy/paste of before. Not sure why.

You're saying you're getting a different error with and without bitdefender enabled?

**Reply by tubefan** [reply]
I'll check it again. But first, please reactivate the license. (see above > GrampaWildWilly)

**Reply by GrampaWildWilly** [reply]
Click the Recheck license button on the first page of the VDH settings.  That should send you an E-mail.  Open that & do what it says there.  That should unlock your license.  You can see instructions for this including screenshots over here https://groups.google.com/g/video-downloadhelper-q-and-a/c/BzPLK2YyL-s.  Expand that so the full text of every post is visible.  Then look for (Find in Page) "locked" within the text of that web page.

**Comment by paulrouget** [comment]
Another thing to try: install that version of the coapp, and tell me if it works any better:

https://github.com/aclap-dev/vdhcoapp/releases/tag/v2.0.10

Thanks again for taking the time to test the beta :)

**Comment by tubefan** [comment]
Your eyes are OK - you're absolutely right! I think the video is great, as are the guys themselves. By the way: Keith is said to have not only stopped smoking but also drinking (at least whiskey).

 

Von: GrampaWildWilly ***@***.***> 
Gesendet: Freitag, 2. Februar 2024 13:29
An: aclap-dev/video-downloadhelper ***@***.***>
Cc: tubefan ***@***.***>; Author ***@***.***>
Betreff: Re: [aclap-dev/video-downloadhelper] Can't download from YOUTUBE anymore (Discussion #14)

 

My eyes might be playing tricks on me but I could swear I saw Bill Wyman in that video. There seemed to be a lot of footage of the very much younger versions of the guys, from at least 40 years ago. There's one point where it shows Keith sticking a cigarette in his mouth. Seems to me I read recently that he had quit smoking.

**Reply by GrampaWildWilly** [reply]
@tubefan I wanted to close this but it seems I can't.  You opened this so I think you need to close it.  Please mark my summary below as the answer & close this discussion as answered.

**Comment by GrampaWildWilly** [comment]
OK.  There's a lot of thrashing about in this thread & it needs a concise summary.

The problem here was a result of a false positive in Bitdefender.  This was reported a couple of weeks ago on our old Google forum but it's good that we now have this here in our new GitHub forum.  I'm sure the Bitdefender folks will get around to correcting Bitdefender so it doesn't throw up a false positive on the CoApp.  In the meantime, you need to make an exception for the CoApp in both the antivirus component of Bitdefender & in the firewall component.  There are a few .exe files in the CoApp directory.  They must all be identified to both components of Bitdefender as exceptions.  It might be easier to just make an exception for the CoApp directory.  That directory can be anywhere on your system.  You are given the chance to accept the default directory or change it during installation of the CoApp.  If you are unsure of where that location is, just execute the CoApp installer again.  Don't actually reinstall the CoApp; that's not necessary, although it won't harm anything.  Just launch the installer, go through the first 2 screens, then pay attention to what it tells you on the third screen:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/bb290848-44af-4b13-b270-f6201625ce55)

The Destination Folder is where your CoApp is located.  That's the one that requires exceptions in Bitdefender.  In this example, it shows a directory that is not the default.  You must use whatever it shows you on your system.  Once you have that information, just cancel this install.

This approach is NOT unique to Bitdefender.  ANY antivirus program might throw a false positive on the CoApp.  You must make the same sort of exceptions in whatever antivirus & firewall you might be running.  That might be one application, it might be two.  Do whatever is appropriate on your system.  Not everybody will encounter this problem.  For example, I use the antivirus provided by Microsoft built into Windows 11.  It has not thrown any false positives so I have not had to make the exceptions I describe here.  Each system is different.  Do what is appropriate on your system.

**Reply by tubefan** [reply]
Hi Paul & GrampaWildWIlly,
thank you very much again for your endless patience and advice!
All the best to you
